packages/webview_flutter/lib/webview_flutter.dart

```diff
@@ -151,6 +151,10 @@ typedef void PageLoadingCallback(int progress);
 /// Signature for when a [WebView] has failed to load a resource.
 typedef void WebResourceErrorCallback(WebResourceError error);
 
+/// Signature for when a [WebView] is requesting authentication.
+typedef WebViewAuthInfo OnReceivedHttpAuthRequestCallback(
```

Comment:
Is there any reason this has to be synchronous? Wouldn't it make more sense to have this `async` (ie. returning `Future<WebViewAuthInfo>`) I could imagine that retrieving credentials (or asking the user) will not always be possible synchronously.

Reply:
Yes, that's what I also thought. I will change it to be async.

Reply:
👍️ just discovered that it should probably be `Future<WebViewAuthInfo?>` return type (ie. nullable), right now the callback has no way of indicating that it has no auth info available?